apps/extension/client/package.json

```diff
@@ -71,6 +71,11 @@
           "type": "boolean",
           "default": true,
           "description": "Show warnings that prevent you from using the visual editor."
+        },
+        "xstate.targetEditorBaseUrl": {
+          "type": "string",
+          "default": "https://stately.ai",
+          "description": "The base URL to open the visual editor in."
```

Comment:
This makes it a public option and I think that it should be avoided. Couldn't you source this from a local configuration file or an environment variable to make this at least more hidden from the users?

Reply:
Why do you think this should be avoided? I think it's fine to have it public. It makes it ultra-convenient in development.

Reply:
How an external user can really use this? How useful it is for them to see this in autocomplete when configuring the extension? It's just that IMHO if it's not useful for them then it shouldn't be there -> fewer options === less mental overhead.

Would a different place for configuring this locally be cumbersome to use?

Reply:
This is a typical place to put the setting IMO - other extensions do similar (these are from mine):

![CleanShot 2022-03-18 at 07 47 17](https://user-images.githubusercontent.com/1093738/158997746-a2b35c6d-d4fe-4cdd-8cb0-b394111970e2.png)
.

Reply:
Right, this is a typical place to put a setting - I don't argue that. I just say that this is not a user-facing setting and as such it probably shouldn't be in such generated forms etc.

Reply:
In the future, we could use this to cater to users who want to self-host the editor/registry, or have it on some stately subdomain (`beta.stately.ai`) or other user-facing ideas.

Reply:
Isnt this potentially YAGNI right now? It's not hard to introduce this later after all, when needed. 

To clarify - this is a nit-level and of course just my opinion, feel free to ignore. Just from my PoV this kinda becomes the API surface when we make this a public option but I dont see how a regular user would use this *today* and therefore my preference is to hide this from this public API surface

Reply:
It is potentially YAGNI, but it's very very useful in development and has little cost to add, so I figured I'd add it.

Reply:
But I'm not saying that we shouldn't have this configurable - just that perhaps a different way of configuring this during dev would be almost as good as this one and that an alternative way wouldn't expose this to our users, where I consider the latter to be a "hidden cost" of adding this as a public option.

Reply:
It's hard to imagine a better option than this during development - you can even change it while the server is running, or specify it differently in different repos with workspace-based config.